### DISCUSSION TITLE
Kicking User/Ending Session to get Reaper Recording File

### DISCUSSION THREAD

**Discussion by brandflake11** [jamulussoftware/jamulus#2928] [Technical Support & Help]
Hello all,

I know there has been a lot of discussion on the topic of kicking users and the general sentiment is that it's not needed. Right now, we are using Jamulus to record sessions, but there seems to be some ghost/bots that are constantly connected to public servers in the "Any Genre 1" room. Is there any way to end the session peacefully so that we can get the .rpp reaper file? I don't want to quit the server, as I'm afraid we will lose the session, and I thought about just disconnecting the computer from the internet, but I don't want to crash the server. 

Because of this, I will definitely get a static ip and run a private server now, but I want to get this session recording.

**Comment by hoffie** [comment]
> Is there any way to end the session peacefully so that we can get the .rpp reaper file?

Can you share some more details about your setup (GUI/headless, OS)?

- For GUI servers there should be a button to end the recording :)
- For headless *nix servers, there's the USR2 option: https://jamulus.io/wiki/Running-a-Server#controlling-recording
- Since 3.9.0 there is an experimental JSON-RPC interface which also supports a [stopRecording](https://github.com/jamulussoftware/jamulus/blob/master/docs/JSON-RPC.md#jamulusserverstoprecording) method which should work on all platforms / deployment types.

**Reply by brandflake11** [reply]
Yes, this is with a version 3.9.0 of Jamulus that I compiled myself, on Ubuntu Studio 22.04.1 LTS. The server is currently running headless. I also don't use systemd to control the Jamulus Server, I just start a script in a screen session.

To be clear, if I go into htop and send the SIGUSR1 signal to the Jamulus server process, it will stop the recording and start a new one, saving my reaper project? I just want to make absolutely sure before I do it to not lose any recording. :P

The JSON-RPC looks very interesting, that's something I'll definitely try to play with in the future.

**Reply by hoffie** [reply]
> I also don't use systemd to control the Jamulus Server, I just start a script in a screen session.

SIGUSR1/2 should work independent of systemd.

> To be clear, if I go into htop and send the SIGUSR1 signal to the Jamulus server process, it will stop the recording and start a new one, saving my reaper project? I just want to make absolutely sure before I do it to not lose any recording. :P

To be honest, I'm not a heavy user of the recording feature. But yes, that's what is documented and what should work. To be extra sure, you can test it with a second Jamulus instance. You can also create a copy of the recording directory. There's also a script by @pljones  which helps recover aborted recordings (https://github.com/pljones/jamulus-jamexporter#recover-recordingsh).

> Yes, this is with a version 3.9.0 of Jamulus that I compiled myself, on Ubuntu Studio 22.04.1 LTS.

Just curious, what is the reason for compiling yourself instead of using the provided debs? :)

**Reply by brandflake11** [reply]
@hoffie Thank you very much for your detailed help, the SIGUSR1 does indeed trigger the recording to end and a new one to begin. Also, that script could be a lifesaver, thank you for that as well. 

> Just curious, what is the reason for compiling yourself instead of using the provided debs? :)

I compile myself so I can have control over the build. I don't know if installing the .deb keeps Jamulus up to date by installing a repo, but I want to make sure that I can easily stay on a version if I so want to. It also allows me to make patches to the Jamulus build, if needed. I used to have to apply a patch to increase the max number of users a Jamulus server could support. And, it's dead simple to compile the binary myself, and I figure that having it compiled on my machine might keep it more stable, as everything will be according to my machine's libraries. This is just what I imagine, I don't know if it actually helps with these things, but it's also fun to compile the build myself.
